src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/PropagateCancellationTokensWhenPossibleTests.cs

```diff
@@ -4,11 +4,266 @@
 using Microsoft.CodeQuality.VisualBasic.Analyzers.ApiDesignGuidelines;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Test.Utilities;
+using Xunit;
 
 namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
 {
     public class PropagateCancellationTokensWhenPossibleTests : DiagnosticAnalyzerTestBase
     {
+        [Fact]
+        public void WipCSharp_CancellationTokenOverloadAvailable()
+        {
+            var source = @"
+using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    Task M1() => throw null;
+
+    Task M1(CancellationToken p1) => throw null;
+
+    Task M2(CancellationToken p1) => M1();
+}
+";
+            VerifyCSharp(source, GetCSharpResultAt(11, 38));
+        }
+
+        [Fact]
+        public void WipCSharp_OptionalCancellationTokenParameter()
+        {
+            var source = @"
+using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    Task M1(CancellationToken p1 = default(CancellationToken)) => throw null;
+
+    Task M2(CancellationToken p1) => M1();
+}
+";
+            VerifyCSharp(source, GetCSharpResultAt(9, 38));
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn-analyzers/issues/1476")]
```

Comment:
What exactly isn't working about these tests? Can you please add info to the linked bug? You explain where the fix needs to happen, but not why it's not working the first place.

Reply:
Done